### PR TITLE
Feat/existing participant view

### PR DESCRIPTION
**Pull request by MikeyManoguerra**
Sets up a nested router for the various visit options needed for a single participant.

It could use better testing. 
I need to do more reading on how to set a mobx state in the test ( for participant Info). 
I want to get feedback on its current implementation, and then either return to test it or revisit later

### CODE REVIEW

File: frontend/src/components/VisitRouter/VisitForm.js [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+            required
+            name="program"
+            value={visitInfo.program}
+            onChange={e => setVisitProgram(e.target.value)}
+            labelId="program-select"
+          >
+            {programList.map(program => (
+              <MenuItem
+                key={program.id}
+                value={programList && programList.length > 0 ? program.id : ""}
```
Comment (ThrowsException) [inline frontend/src/components/VisitRouter/VisitForm.js:45]: programList has to be > 0 for this to display so these checks aren't necessary

File: frontend/src/components/VisitRouter/VisitForm.js [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+              name="service"
+              value={visitInfo.service}
+              onChange={e => setVisitService(e.target.value)}
+              labelId="service-select"
+            >
+              {serviceList.map(service => (
+                <MenuItem
+                  key={service.id}
+                  value={
+                    serviceList && serviceList.length > 0 ? service.id : ""
```
Comment (ThrowsException) [inline frontend/src/components/VisitRouter/VisitForm.js:68]: same here. serviceList will have to exist and be greater than 0 or the map doesn't run at all.

Comment (MikeyManoguerra) [reply frontend/src/components/VisitRouter/VisitForm.js:68]: ok can do

File: frontend/src/components/ParticipantForm.js [hunk 3]
```diff
@@ -64,7 +69,7 @@ function ParticipantForm(props) {
         />
```
Comment (LizKalter) [inline frontend/src/components/ParticipantForm.js:69]: Did you want to add the disabled attribute to the dob field as well?

Comment (MikeyManoguerra) [reply frontend/src/components/ParticipantForm.js:69]: oops yes i guess so!

**Comment by MikeyManoguerra** [comment]
@LizKalter I fixed This PR. the "exact" flag snuck back in on the parent router when I rebased. I obviously did not retest!

**Comment by MikeyManoguerra** [comment]
@LizKalter on rebase, the "exact" flag snuck back in on the nested router!  Trying to do the merge without re-checking was  a fail sry.

File: frontend/.babelrc [hunk 4]
```diff
@@ -3,6 +3,6 @@
   "plugins": [
     ["@babel/plugin-transform-runtime", { "regenerator": true }],
     ["@babel/plugin-proposal-decorators", { "legacy": true }],
-    ["@babel/plugin-proposal-class-properties", { "loose": true }]
+    ["@babel/plugin-proposal-class-properties", { "loose": true }, "classes-alias"]
```
Comment (ThrowsException) [inline frontend/.babelrc:6]: I think we can take this out now right?

File: frontend/src/components/VisitRouter/VisitData.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+            the api route would be
+            'api/visit/:visit_id/populated/'
+
+            the front end route may also benefit from id based path
+            thus being '/existingParticipant/:participant_id/visits/:visit_id/'
+
+            This Component is not very well defined yet, and could be come to different routes, such as:
+            '/existingParticipant/:participant_id/visits/:visit_id/forms'
+            '/existingParticipant/:participant_id/visits/:visit_id/data'
+            */}
```
Comment (ThrowsException) [inline frontend/src/components/VisitRouter/VisitData.js:31]: rather see this be a doc string above the function than a comment in it

File: frontend/src/components/VisitRouter/VisitForm.js [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+import Grid from "@material-ui/core/Grid"
+import Select from "@material-ui/core/Select"
+import MenuItem from "@material-ui/core/MenuItem"
+import TextField from "@material-ui/core/TextField"
+import InputLabel from "@material-ui/core/InputLabel"
+import FormControl from "@material-ui/core/FormControl"
+import { PrevPointHeading } from "../Typography"
+import { URGENCY_OPTIONS } from "../../constants"
+
+function VisitForm(props) {
```
Comment (ThrowsException) [inline frontend/src/components/VisitRouter/VisitForm.js:12]: Any reason we aren't using const and arrow functions for these?
`const VisitForm = (props) => {...}`

File: frontend/src/components/VisitRouter/VisitForm.js [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+  const { visitInfo, programList, serviceList, handleVisitChange } = props
```
Comment (ThrowsException) [inline frontend/src/components/VisitRouter/VisitForm.js:13]: if we know this is what we need out of props why not just make it destrcuture in the function
`const VisitForm = ({visitInfo, programList, serviceList, handleVisitChange}) = {...}`

File: frontend/src/components/VisitRouter/VisitForm.js [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+                {program.name}
+              </MenuItem>
+            ))}
+          </Select>
+        </FormControl>
+      </Grid>
+      <Grid item xs={12} sm={6}>
+        <FormControl>
+          <InputLabel id="service-select">Select Service</InputLabel>
+          {visitInfo.program && serviceList.length > 0 ? (
```
Comment (ThrowsException) [inline frontend/src/components/VisitRouter/VisitForm.js:45]: We don't need to check length > 0 here

File: frontend/src/components/VisitRouter/VisitTable.js [hunk 9]
```diff
@@ -0,0 +1,30 @@
+import React from "react"
+import { Link } from "react-router-dom"
+import Grid from "@material-ui/core/Grid"
+import Container from "@material-ui/core/Container"
+import { PrevPointCopy } from "../Typography"
+import PrevPointButton from "../PrevPointButton"
+
+export default function VisitTable() {
```
Comment (ThrowsException) [inline frontend/src/components/VisitRouter/VisitTable.js:8]: Im kind of curious about a component with no props. why use this?

File: frontend/src/views/ExistingParticipantView.js [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+    // kick off api calls for program list from Mobx
+    participantStore.getPrograms()
+    // -----------------------------------
+
+    // return to /participants page if "current participant" is not set
+    // add participants in NewParticipantView
+    if (!Object.keys(participantStore.participant).length) {
+      history.push("/participants")
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (ThrowsException) [inline frontend/src/views/ExistingParticipantView.js:57]: We should really stop doing this =(

File: frontend/.babelrc (hunk 4 above)
Comment (MikeyManoguerra) [reply frontend/.babelrc:6]: Your  PR has not been  merged yet

File: frontend/src/components/VisitRouter/VisitForm.js (hunk 6 above)
Comment (MikeyManoguerra) [reply frontend/src/components/VisitRouter/VisitForm.js:12]: My personal preference seeping in, Ill switch them back to stay consistent

File: frontend/src/components/VisitRouter/VisitTable.js (hunk 9 above)
Comment (MikeyManoguerra) [reply frontend/src/components/VisitRouter/VisitTable.js:8]: Its just a stub, setting up for props to be a participants list of visits. that  api endpoint doesnt  exist yet.

**Pull request merged**